transport_nantes/mobilito/views.py

```diff
@@ -436,3 +452,50 @@ def get(self, request: HttpRequest, *args, **kwargs) -> HttpResponse:
         context = self.get_context_data(**kwargs)
         self.check_view_permission(context["mobilito_session"])
         return self.render_to_response(context)
+
+
+@csrf_protect
+def report_session(request: HttpRequest, **kwargs) -> HttpResponse:
+    """Report a session as inappropriate"""
+
+    if request.method == 'POST':
+        if request.user.is_authenticated:
+            user = request.user
+        else:
+            user = None
+
+        session_sha1 = kwargs['session_sha1']
+        try:
+            # In the event that somehow, in spite of CSRF protection, a user
+            # tries to report a session that doesn't exist, we don't want to
+            # crash the app, so we catch the exception, log it and return a 404
+            session_object = get_object_or_404(
+                Session, session_sha1=session_sha1)
+        except Http404:
+            logger.error(f"{user or 'Anonymous User'} tried to report a "
+                         "non-existing session")
+            raise Http404
+        if session_object:
```

Comment:
Under what circumstances could this be False?

Under those circumstances, you've decided you can silently ignore the case.  Why?